### PR TITLE
[HB-5107] 3rd round listener fixes and hardening

### PR DESCRIPTION
**Pull request by bichenwang**

### CODE REVIEW

File: YahooAdapter/build.gradle.kts [hunk 1]
```diff
@@ -68,7 +68,7 @@ dependencies {
 
     // For external usage, please use the following production dependency.
     // You may choose a different release version.
-    "remoteImplementation"("com.chartboost:chartboost-mediation-sdk:4.+")
+    "remoteImplementation"("com.chartboost:chartboost-mediation-sdk:4.0.0")
```
Comment (CB-RyanMcCormick) [inline YahooAdapter/build.gradle.kts:71]: 4.0.0?

**Review by CB-RyanMcCormick — APPROVED** [review]
One question, LGTM otherwise

File: YahooAdapter/build.gradle.kts (hunk 1 above)
Comment (bichenwang) [reply YahooAdapter/build.gradle.kts:71]: yep 4.0.0! (it will get moved up if you personally specify a higher version)

**Review by cb-jpadilla — APPROVED** [review]
✔️

**Review by chauduyphanvu — APPROVED** [review]

**Pull request merged**